internal/application/command.go

```diff
@@ -451,6 +451,11 @@ func validateServiceAndDeviceState(deviceName string, dic *di.Container) (models
 		return models.Device{}, errors.NewCommonEdgeX(errors.KindServiceLocked, fmt.Sprintf("device %s OperatingState is DOWN", device.Name), nil)
 	}
 
+	// check device's ProfileName
+	if device.ProfileName == "" {
+		return models.Device{}, errors.NewCommonEdgeX(errors.KindContractInvalid, "no associated device profile", nil)
```

Comment:
```suggestion
		return models.Device{}, errors.NewCommonEdgeX(errors.KindServiceLocked, "no associated device profile", nil)
```

since the empty profile is valid, we should not return 400 error
please return 423 instead and update the swagger file to describe this possibility